src/services/url.ts

```diff
@@ -22,9 +22,25 @@ class UrlService extends BaseService<UrlService.Options> {
   }
 
   init() {
+    this.beautifier = new UrlBeautifier(this.generateRoutes(), this.opts.beautifier, this.app.config);
     this.readInitialUrl();
   }
 
+  generateRoutes() {
+    const baseTag = WINDOW().document.head.querySelector('base');
+    const routes = this.opts.routes;
+    let baseUrl = baseTag && baseTag.hasAttribute('href') ? baseTag.getAttribute('href') : '/';
+    const position = baseUrl.length - 1;
+    baseUrl = baseUrl.charAt(position) === '/' ? baseUrl.substring(0, position) : baseUrl;
+    let generatedRoutes: any = {};
```

Comment:
this can be better represented with a `reduce()` statement

```js
return Object.keys(routes)
  .reduce((generatedRoutes) => Object.assign(generatedRoutes, { [prop]: baseUrl + routes[prop] }), {});
```